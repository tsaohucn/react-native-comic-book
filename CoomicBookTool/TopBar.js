import React from 'react'
import { 
  View, 
  StyleSheet, 
  Text, 
  Animated, 
  Dimensions, 
  TouchableWithoutFeedback 
} from 'react-native'
import PropTypes from "prop-types"
import { Icon } from 'react-native-elements'

const TopBar = ({ title, animatedTopBarY, onClickBackArrow, onClickOptionBar }) => (
  <Animated.View style={[styles.topToolBarAnimated,{
    transform: [
      { translateY: animatedTopBarY  }
    ]}]}
  >
    <TouchableWithoutFeedback
      onPress={onClickBackArrow} 
    >
      <View style={styles.topToolBarIconView}>
        <Icon
          iconStyle={styles.topToolBarIcon}
          name='arrow-left'
          type='material-community'
          color='white'
        />
      </View>
    </TouchableWithoutFeedback>
    <Text numberOfLines={1} style={styles.iconText}>{title}</Text>
    <TouchableWithoutFeedback
      onPress={onClickOptionBar} 
    >
      <View style={styles.topToolBarView}>
        <Icon
          iconStyle={styles.topToolBarIcon}
          name='dots-horizontal'
          type='material-community'
          color='white'
        />
      </View>
    </TouchableWithoutFeedback>
  </Animated.View>
)

const { width, height } = Dimensions.get('window')

const styles = StyleSheet.create({
  iconText: {
    width: width*0.7,
    color: 'white'
  },
  topToolBarAnimated: {
    position: 'absolute',
    top: 0,
    width,
    height: 50,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',    
    backgroundColor: 'rgba(52, 52, 52, 0.8)'
  },
  topToolBarIconView: {
    height: 50,
    justifyContent: 'center'
  },
  topToolBarIcon: {
    paddingLeft: 10,
     paddingRight: 10
  }
})

TopBar.propTypes = {
  title: PropTypes.string,
  animatedTopBarY: PropTypes.instanceOf('Animated'),
  onClickBackArrow: PropTypes.func,
  onClickOptionBar: PropTypes.func
};

TopBar.defaultProps = {
  animatedTopBarY: 0
};

export default TopBar