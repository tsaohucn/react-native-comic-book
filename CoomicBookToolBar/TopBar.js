import React from 'react'
import { 
  View, 
  StyleSheet, 
  Text, 
  Animated, 
  Dimensions, 
  TouchableWithoutFeedback 
} from 'react-native'
import { Icon } from 'react-native-elements'

const TopBar = ({ animatedTopBarY, onClickBackArrow, onClickOptionBar }) => (
  <View>
  <Animated.View style={[styles.topToolBarAnimated,{
    transform: [
      { translateY: animatedTopBarY  }
    ]}]}
  >
    <TouchableWithoutFeedback
      onPress={onClickBackArrow} 
    >
      <View style={styles.topToolBarIconView}>
        <Icon
          iconStyle={styles.topToolBarIcon}
          name='arrow-left'
          type='material-community'
          color='white'
        />
      </View>
    </TouchableWithoutFeedback>
    <Text style={styles.iconText}>{'第一話'}</Text>
    <TouchableWithoutFeedback
      onPress={onClickOptionBar} 
    >
      <View style={styles.topToolBarView}>
        <Icon
          iconStyle={styles.topToolBarIcon}
          name='dots-horizontal'
          type='material-community'
          color='white'
        />
      </View>
    </TouchableWithoutFeedback>
  </Animated.View>
  </View>
)

const { width, height } = Dimensions.get('window')

const styles = StyleSheet.create({
  iconText: {
    color: 'white'
  },
  topToolBarAnimated: {
    position: 'absolute',
    top: 0,
    width,
    height: 50,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',    
    backgroundColor: 'rgba(52, 52, 52, 0.8)'
  },
  topToolBarIconView: {
    height: 50,
    justifyContent: 'center'
  },
  topToolBarIcon: {
    paddingLeft: 10,
     paddingRight: 10
  }
})

export default TopBar