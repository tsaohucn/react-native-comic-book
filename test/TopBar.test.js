// __tests__/TopBar.js
import 'react-native'
import React from 'react'
import TopBar from '../CoomicBookTool/TopBar'

import renderer from 'react-test-renderer'

jest.mock('react-native-elements', () => {
	return {
		Icon: 'Icon'
	}
})

describe('Testing TopBar component', () => {

	it('renders correctly with no props', () => {
		const props = {

		}
	  const tree = renderer.create(
	    <TopBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

	it('renders correctly with animatedTopBarY value', () => {
		const props = {
			animatedTopBarY: 100
		}
	  const tree = renderer.create(
	    <TopBar {...props}/>
	    ).toJSON();
	  expect(tree).toMatchSnapshot();
	})

})