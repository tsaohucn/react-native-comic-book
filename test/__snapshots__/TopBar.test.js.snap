// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TopBar component renders correctly with animatedTopBarY value 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "position": "absolute",
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 100,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      iconStyle={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      name="arrow-left"
      type="material-community"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }
  >
    第一話
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <Icon
      color="white"
      iconStyle={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      name="dots-horizontal"
      type="material-community"
    />
  </View>
</View>
`;

exports[`Testing TopBar component renders correctly with no props 1`] = `
<View
  collapsable={undefined}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(52, 52, 52, 0.8)",
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "space-between",
      "position": "absolute",
      "top": 0,
      "transform": Array [
        Object {
          "translateY": 0,
        },
      ],
      "width": 750,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "height": 50,
        "justifyContent": "center",
      }
    }
    testID={undefined}
  >
    <Icon
      color="white"
      iconStyle={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      name="arrow-left"
      type="material-community"
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }
  >
    第一話
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={undefined}
    testID={undefined}
  >
    <Icon
      color="white"
      iconStyle={
        Object {
          "paddingLeft": 10,
          "paddingRight": 10,
        }
      }
      name="dots-horizontal"
      type="material-community"
    />
  </View>
</View>
`;
